salt/modules/junos.py

```diff
@@ -220,8 +220,10 @@ def rpc(cmd=None, dest=None, **kwargs):
         return ret
 
     format_ = op.pop("format", "xml")
-    # when called from state, dest becomes part of op via __pub_arg
-    dest = dest or op.pop("dest", None)
+    # dest becomes part of op via __pub_arg if not None
+    # rpc commands opjecting to dest as part of op
+    if dest:
+        op.pop("dest", None)
```

Comment:
this need change. op will contain dest if called via state file.
so it should be 

if dest is None:
    dest = op.pop("dest", None)

or we can also do
dest = op.pop("dest", dest)

Reply:
Found that if dest is specified via a state file or execution module that dest is there as a keyword.